### ISSUE TITLE
Display US definition of quartiles in Statistics

### ISSUE THREAD

**Issue by artaxxx** [numworks/epsilon#158]
As we want to avoid to add this in Settings, the best way should be to use the US definition of quartiles when the language is English and use the other definition for the rest.

**Comment by madhatter0** [comment]
Is it possible to understand the backstory here a little better?  *If* there are different definitions of quartile *and* one is used in the US while another is used in the UK (I have no idea if this is true, which is why I'd like to understand the backstory better) *then* using the US one when the language is set to English is not helpful.

**Comment by adriweb** [comment]
I agree that in such cases, grouping together language and math definition isn't the best idea. What if India, UK, Australia, and US schools aren't taught the same definitions? Despting students choosing English, they'd need to be able to choose whichever setting they need!

**Comment by artaxxx** [comment]
The problem in this case is that students cannot chose between different definitions because they do not know them. It's even hard for us to select our definition (I didn't know the existence of the US definition till last year...). I think we shouldn't allow the user to choose between several quartiles definitions. 

I agree with you that it's a little weird to link language selection with math definitions but we can simply add an "English US" version to the language selection screen.
Does anyone know if the use of the US definition is limited to the US?

**Comment by adriweb** [comment]
Better idea, then: instead of letting the user choose the "US definition" or "French definition", just select by default something according to the current language, AND have a settings submenu where the user could choose a definition of things like quartile **from a list of formulas** (or whatever it is).
The user then wouldn't choose _"French definition"_, but _"Q1: <= 25%"_, etc. People wouldn't need to know which definition it is for which country. Much more visual and intuitive IMHO.

**Comment by ian-abbott** [comment]
Casio Prizm calculators have a "Q1Q3 Type" setting that lets you choose between two methods.

**Comment by madhatter0** [comment]
Artaxxx, it'd be helpful if you could briefly summarise how it's done in the US.  Locals can then confirm whether or not it's done that way in other English-speaking countries.

**Comment by artaxxx** [comment]
Good suggestion @madhatter0 !

Let's do that with a simple set of data: {1,2,3,4,5}.

In the US, Q1=1.5 and Q3=4.5 (you set the median apart, which is 3 and then Q1=med{1,2} and Q3=med{4,5}).

In France, Q1=2 and Q3=4 (rounded up in a way)

**Comment by adriweb** [comment]
Well in France, it's just that the Q1 has to belong to the set, IIRC, and follow the definition: "Q1: <= 25%"

**Comment by ian-abbott** [comment]
The English Wikipedia pages list 3 methods for calculating quartiles. Wolfram MathWorld lists 5 methods!

**Comment by madhatter0** [comment]
OK, I've checked with a friendly local A-level maths teacher.  The textbook they sent me a snippet of notes that "_there are many different methods for calculating the quartiles, and they may give slightly different answers_".  They then go on to say "_this book will use the following method_" and give both a method and an example which is the same as the French method.  So UK students at least do expect there to be several methods of calculating the IQR, and don't always default to the US method.

Let me know if you'd like me to attach the scanned page of textbook (on a fair-use basis).

**Comment by dicorynia** [comment]
There are at least 9 ways to compute quartiles ; see (for R) :

```
> quantile(c(1,2,3,4,5), type = 1)
  0%  25%  50%  75% 100% 
   1    2    3    4    5 
> quantile(c(1,2,3,4,5), type = 2)
  0%  25%  50%  75% 100% 
   1    2    3    4    5 
> quantile(c(1,2,3,4,5), type = 3)
  0%  25%  50%  75% 100% 
   1    1    2    4    5 
> quantile(c(1,2,3,4,5), type = 4)
  0%  25%  50%  75% 100% 
1.00 1.25 2.50 3.75 5.00 
> quantile(c(1,2,3,4,5), type = 5)
  0%  25%  50%  75% 100% 
1.00 1.75 3.00 4.25 5.00 
> quantile(c(1,2,3,4,5), type = 6)
  0%  25%  50%  75% 100% 
 1.0  1.5  3.0  4.5  5.0 
> quantile(c(1,2,3,4,5), type = 7)
  0%  25%  50%  75% 100% 
   1    2    3    4    5 
> quantile(c(1,2,3,4,5), type = 8)
      0%      25%      50%      75%     100% 
1.000000 1.666667 3.000000 4.333333 5.000000 
> quantile(c(1,2,3,4,5), type = 9)
    0%    25%    50%    75%   100% 
1.0000 1.6875 3.0000 4.3125 5.0000 
```
References :
https://stat.ethz.ch/R-manual/R-devel/library/stats/html/quantile.html
https://www.amherst.edu/media/view/129116/original/Sample+Quantiles.pdf

**Comment by ian-abbott** [comment]
Those 9 methods supported by R's `quartile` function are from Hyndman and Fan's 1996 paper "Sample Quantiles in Statistical Packages". It's easy enough to find online, but I won't link to it here because you're supposed to pay for it. Microsoft Excel's `QUARTILE.INC` and `QUARTILE` functions correspond to R's type 6, and Excel's `QUARTILE.EXC` function corresponds to type 7 (ref. [Why Excel has Multiple Quartile Functions and How to Replicate the Quartiles from R and Other Statistical Packages](http://datapigtechnologies.com/blog/index.php/why-excel-has-multiple-quartile-functions-and-how-to-replicate-the-quartiles-from-r-and-other-statistical-packages/)).

**Comment by abhobe** [comment]
When will the quartile update be released?

**Comment by nsajko** [comment]
@abhobe Look at the discussion above: there are more than 9 mutually incompatible definitions in use, so somebody first needs to propose how to present the choice of which one to use to the user, so the Numworks people could decide what to do.

For example, the Numworks people could decide that Epsilon should provide definitions: 1, 2, 3, 5, 7, 8.

Do you know which of the 9 definitions (from the PDF a little bit above, same as in R) do you want?

**Comment by abhobe** [comment]
@nsajko  I am looking for **type 6.** (US Version)

**Comment by artaxxx** [comment]
Fixed in 15.0.0
